app/helpers/argo_helper.rb

```diff
@@ -102,11 +102,11 @@ def render_buttons(doc, object = nil)
     end
 
     # if this is an apo and the user has permission for the apo, let them edit it.
-    if (object.datastreams.include? 'roleMetadata') && (current_user.is_admin || current_user.is_manager || object.can_manage_item?(current_user.roles(apo_pid)))
+    if object.datastreams.include?('roleMetadata') && can?(:manage_item, object)
```

Comment:
Originally only APOs had roles and therefore roleMetadata, and they are still the only roles Argo cares about.  But Hydrus implemented item level roles, so that datastream can appear in items now.  It should just be covered now by any edit/write-level permission for any object. 

Lynn


> On Feb 26, 2016, at 11:40 AM, Chris Beer <notifications@github.com> wrote:
> 
> In app/helpers/argo_helper.rb:
> 
> > @@ -102,11 +102,11 @@ def render_buttons(doc, object = nil)
> >      end
> >  
> >      # if this is an apo and the user has permission for the apo, let them edit it.
> > -    if (object.datastreams.include? 'roleMetadata') && (current_user.is_admin || current_user.is_manager || object.can_manage_item?(current_user.roles(apo_pid)))
> > +    if object.datastreams.include?('roleMetadata') && can?(:manage_item, object)
> 
> My understanding of the following lines is that it only applies to objects that have a roleMetadata datastream. The comment suggests, though, an obvious refactor would be to check if the object is an APO.
> 
> —
> Reply to this email directly or view it on GitHub.
>